public/app/containers/Explore/Table.tsx

```diff
@@ -1,22 +1,58 @@
 import React, { PureComponent } from 'react';
-// import TableModel from 'app/core/table_model';
+import TableModel from 'app/core/table_model';
 
-const EMPTY_TABLE = {
-  columns: [],
-  rows: [],
+const EMPTY_TABLE = new TableModel();
+
+interface TableProps {
+  className?: string;
+  data: TableModel;
+  onClickCell?: (columnKey: string, rowValue: string) => void;
+}
+
+interface SFCCellProps {
+  columnIndex: number;
+  onClickCell?: (columnKey: string, rowValue: string, columnIndex: number, rowIndex: number, table: TableModel) => void;
+  rowIndex: number;
+  table: TableModel;
+  value: string;
+}
+
+const Cell: React.SFC<SFCCellProps> = props => {
```

Comment:
Since we use code style based on Airbnb, I think we should be consistent and use the same way to define components. Should SFC component be defined like a function (`function Cell(props: SFCCellProps) {}`)?
https://github.com/airbnb/javascript/tree/master/react#class-vs-reactcreateclass-vs-stateless

Reply:
I think your right, using named function is usually better as it gives better stack traces

Reply:
Yup, good to agree on this. My style was from [here](https://github.com/piotrwitek/react-redux-typescript-guide#react-types-cheatsheet). Let's go with the airbnb style.